options/posix/generic/pthread.cpp

```diff
@@ -1426,3 +1427,46 @@ int pthread_getcpuclockid(pthread_t, clockid_t *) {
 			<< frg::endlog;
 	return ENOENT;
 }
+
+int pthread_spin_init(pthread_spinlock_t *__lock, int) {
+	__lock->lock = 0;
+	return 0;
+}
+
+int pthread_spin_destroy(pthread_spinlock_t*) {
+	return 0;
+}
+
+int pthread_spin_lock(pthread_spinlock_t *__lock) {
+	unsigned int desired = mlibc::this_tid();
+	unsigned int expected = 0;
+
+	if (__atomic_load_n(&__lock->lock, __ATOMIC_RELAXED) == desired)
```

Comment:
This `if` case can be rolled into the while below, as on failure to compare-and-exchange we get a return value of zero and the lock's value is written into `expected`; there, we could inspect the value of expected instead.